vm/src/stdlib/array.rs

```diff
@@ -464,6 +468,40 @@ fn f64_try_into_from_object(vm: &VirtualMachine, obj: PyObjectRef) -> PyResult<f
     IntoPyFloat::try_from_object(vm, obj).map(|x| x.to_f64())
 }
 
+impl ArrayElement for PyUnicode {
+    fn try_into_from_object(vm: &VirtualMachine, obj: PyObjectRef) -> PyResult<Self> {
+        let ch = PyStrRef::try_from_object(vm, obj)?
+            .as_str()
+            .chars()
+            .nth(0)
```

Comment:
```suggestion
            .next()
```
otherwise if it must be single character, `itertools::exactly_one`

Reply:
Having this check (`exactly_one`) is nice, I raise a `ValueError` instead of panic.

Reply:
Tested with CPython:
```
>>> import array
>>> a = array.array('u', 'abcde')
>>> a.append('cd')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: array item must be unicode character
>>> 
```
It is better to throw a `TypeError`,